fv3core/stencils/d2a2c_vect.py

```diff
@@ -382,72 +379,110 @@ class DGrid2AGrid2CGridVectors:
     Fortran name d2a2c_vect
     """
 
-    def __init__(self, grid, namelist, dord4):
-        if namelist.grid_type >= 3:
-            raise Exception("unimplemented grid_type >= 3")
-        self.grid = grid
+    def __init__(
+        self,
+        grid_indexing: GridIndexing,
+        cosa_s,
+        cosa_u,
+        cosa_v,
+        rsin_u,
+        rsin_v,
+        rsin2,
+        dxa,
+        dya,
+        sin_sg1,
+        sin_sg2,
+        sin_sg3,
+        sin_sg4,
+        nested: bool,
+        grid_type: int,
+        # TODO: remove npx and layout, and in doing so
+        # remove MPI-grid dependence of outputs
+        npx: int,
+        layout: Tuple[int, int],
```

Comment:
In the long run, it's true, we don't want to have a special case of this stencil for c12_54ranks. I think it is not unreasonable for modules to have awareness of the model sizing/rank, but in this case since it is only used for a workaround for small domains, I am ok with us eventually removing it, either by testing with c48 (or c20 or whatever) 54 ranks and raising an error early on in the model, or adapting gt4py or the validation to work in the case of c12_54ranks. I appreciate it not being changed in the scope of this PR>

Reply:
Actually on second thought you could remove these now